lib/cloudware/aws.rb

```diff
@@ -119,32 +121,31 @@ def machines
-                          if test.parameters[0]
-                            @state = 'failed'
-                            @extip = 'N/A'
-                            @instance_id = 'N/A'
-                            test.parameters.each_with_index do |val, index|
-                              @type = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmType'
-                              @domain = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareDomain'
-                              @id = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareId'
-                              @role = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmRole'
-                              @prvip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'prvIp'
-                              @mgtip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'mgtIp'
-                              @flavour = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmFlavour'
-                            end
-                            @name = test.stack_name.gsub("#{@domain}-", '')
-                            log.info("[#{self.class}] Detected machine #{@name} in domain #{@domain}")
-                            @machines.merge!("#{@domain}-#{@name}" => {
-                                               name: @name, domain: @domain, state: @state,
-                                               id: @id, type: @type, role: @role, mgt_ip: @mgtip,
-                                               prv_ip: @prvip, ext_ip: @extip, provider: 'aws', instance_id: @instance_id,
-                                               flavour: @flavour
-                                             })
+                          next unless test.stack_status.include?('ROLLBACK')
+                          next unless test.parameters[0]
+                          @state = 'failed'
+                          @extip = 'N/A'
+                          @instance_id = 'N/A'
+                          test.parameters.each_with_index do |val, _index|
+                            @type = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmType'
+                            @domain = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareDomain'
+                            @id = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareId'
+                            @role = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmRole'
+                            @prvip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'prvIp'
+                            @mgtip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'mgtIp'
+                            @flavour = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmFlavour'
                           end
+                          @name = test.stack_name.gsub("#{@domain}-", '')
```

Comment:
Same issue as mentioned here https://github.com/alces-software/cloudware/pull/32#discussion_r187367650 (I presume it will be addressed here via a rebase)

Reply:
Don't worry about readdressing previous comments, they will be cleared up in the previous PR